data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstRange.java

```diff
@@ -30,8 +36,60 @@
      * @param networkElement: PST element to modify
      */
     @JsonCreator
-    public PstRange(@JsonProperty("networkElement") NetworkElement networkElement) {
-        super(networkElement);
+    public PstRange(@JsonProperty("id") String id,
+                    @JsonProperty("name") String name,
+                    @JsonProperty("operator") String operator,
+                    @JsonProperty("usageRules") List<UsageRule> usageRules,
+                    @JsonProperty("ranges") List<Range> ranges,
+                    @JsonProperty("networkElement") NetworkElement networkElement) {
+        super(id, name, operator, usageRules, ranges, networkElement);
+    }
+
+    public PstRange(String id,
+                    NetworkElement networkElement) {
+        super(id, networkElement);
+    }
+
+    @Override
+    public void synchronize(Network network) {
+        TwoWindingsTransformer transformer = network.getTwoWindingsTransformer(networkElement.getId());
+        PhaseTapChanger phaseTapChanger = transformer.getPhaseTapChanger();
+        int highTapPosition = phaseTapChanger.getHighTapPosition();
+        int lowTapPosition = phaseTapChanger.getLowTapPosition();
+        RangeDefinition rangeDefinition;
+        if (lowTapPosition <= 0) {
+            rangeDefinition = RangeDefinition.CENTERED_ON_ZERO;
+        } else {
+            rangeDefinition = RangeDefinition.STARTS_AT_ONE;
+        }
+        Range absoluteFixedRange = new Range(lowTapPosition, highTapPosition, RangeType.ABSOLUTE_FIXED, rangeDefinition);
+        addRange(absoluteFixedRange);
```

Comment:
I don't understand this part. I thought that the RangeDefinition was something intrinsic to the Crac, that allows the "Crac users" to say whether its input data are given with the CENTERED_ON_ZERO or STARTS_AT_ONE convention. The Range should not be updated by the PstRange.

To discuss tomorrow :)

Reply:
Indeed, instead of creating a Range, we will load the value of TwoWindingsTransformer... low tap position and adapt the method getExtremumValueWithRange to take it into account. Then I will implement the desynchronize method